airflow/providers/google/cloud/operators/bigquery_dts.py

```diff
@@ -359,7 +369,18 @@ def _wait_for_transfer_to_be_done(self, run_id: str, transfer_config_id: str, in
         if interval <= 0:
             raise ValueError("Interval must be > 0")
 
+        idx = 0
         while True:
+            current_tick_div, current_tick_mod = divmod(idx * interval, self.token_refresh_interval_seconds)
+            next_tick_div, next_tick_mod = divmod((idx + 1) * interval, self.token_refresh_interval_seconds)
+            if (current_tick_div < next_tick_div and 0 < next_tick_mod) or (
+                current_tick_mod == 0 and 1 <= current_tick_div
+            ):
+                _ = self.hook.refresh_credentials()
+                self.log.info(
+                    f"Credentials were refreshed on tick: idx={idx}, idx*interval={idx * interval} sec"
+                )
```

Comment:
Other providers also requires refresh logic but seems like they have simpler implementation. For example:
https://github.com/apache/airflow/blob/0b680c94922e3f7ca1f3ada8328e315bbae37dc8/airflow/providers/databricks/hooks/databricks_base.py#L441-L453

Can you explain why we need complex logic here?

Also please extract the check validation (time handling) to it's own private function - it should be covered by unit tests

Reply:
Thank you for quick and detailed review..!

Actually, I can't find the token validation method on `base_google.py` module, as like `databricks_base.py` module you mentioned.. If it has similar method based on Google sdk, it will be more simple and elegant way, as you said. (But, it tasks more GCP API calls to check token validity on event loop)

We experienced some error cases, even though the tasks take just over 30 minutes, not 1 hour, with same authorization error. (I don't know why..) so I decided to use the half of default token lifespan to refresh token.